src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -2089,7 +2090,8 @@ static class PutDirectoryException extends IOException {
     private final List<Throwable> exceptions;
 
     PutDirectoryException(Path path, Digest digest, List<Throwable> exceptions) {
-      super(String.format("%s: %d exceptions", path, exceptions.size()));
+      // When printing the exception, show the captured sub-exceptions.
+      super(String.format("%s: %d exceptions: %s", path, exceptions.size(), exceptions));
```

Comment:
While you're at it, can you fix up the plural here? ChoiceFormat almost looks appropriate for this, if overkill

Reply:
Using a simple ternary for singular/plural now.